src/meshes/t8code_mesh.jl

```diff
@@ -731,9 +731,14 @@ function adapt!(mesh::T8codeMesh, adapt_callback; recursive = true, balance = tr
 
         t8_forest_set_ghost(new_forest, ghost, T8_GHOST_FACES) # Note: MPI support not available yet so it is a dummy call.
 
+        # Julias's GC leads to random segfaults here. Temporarily switch it off.
+        GC.enable(false)
+
         # The old forest is destroyed here.
         # Call `t8_forest_ref(Ref(mesh.forest))` to keep it.
         t8_forest_commit(new_forest)
+
+        GC.enable(true)
```

Comment:
Do you have an idea why? Or was this something you tried along the way but is not required given the other fixes?

Reply:
@ranocha No, unfortunately I do not exactly know why. This fix is not directly related to the finalizer calls added to the elixirs in this PR.

But this fix prevents random segfaulting when doing mesh adaptations. There seems to be some complex interaction between the Julia runtime and t8code which I do not fully understand. The garbage collector deletes some objects which are still used by t8code.

Reply:
Ok, thanks